File: libsignal-service/src/main/java/org/whispersystems/signalservice/api/subscriptions/StripeClientSecret.java
package org.whispersystems.signalservice.api.subscriptions;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public final class StripeClientSecret {

  private final String id;
  private final String clientSecret;

  @JsonCreator
  public StripeClientSecret(@JsonProperty("clientSecret") String clientSecret) {
    this.id           = clientSecret.replaceFirst("_secret.*", "");
    this.clientSecret = clientSecret;
  }

  public String getId() {
    return id;
  }

  public String getClientSecret() {
    return clientSecret;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/api/subscriptions/StripeClientSecret.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.api.subscriptions

data class StripeClientSecret(
  val clientSecret: String
) {
  val id: String
    get() = clientSecret.replaceFirst("_secret.*".toRegex(), "")
}
